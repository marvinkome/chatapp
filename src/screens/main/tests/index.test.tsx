import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import MainScreen from '../index';
import View from '../view';

describe('main screen tests', () => {
    it('renders correctly with defaults', () => {
        const app = renderer.create(<MainScreen />).toJSON();
        expect(app).toMatchSnapshot();
    });

    it('renders view correctly with defaults', () => {
        const app = renderer.create(<View />).toJSON();
        expect(app).toMatchSnapshot();
    });
});
