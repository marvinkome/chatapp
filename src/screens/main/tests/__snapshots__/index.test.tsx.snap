// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main screen tests renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 34,
                    "marginLeft": 1,
                    "marginRight": 2,
                    "marginTop": 0,
                    "paddingTop": 1,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            Ôèè
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Main Page
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": 10,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    padder={true}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            avatar={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../static/default-pic.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 28,
                    "height": 56,
                    "width": 56,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "marginLeft": 15,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                James Bond
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                note={true}
                style={
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                Doing what you like will always keep you happy..
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "paddingBottom": 10,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                note={true}
                style={
                  Object {
                    "alignSelf": null,
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "200",
                  }
                }
                uppercase={false}
              >
                0
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            avatar={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": null,
                "paddingRight": null,
                "paddingTop": null,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "flex": 0,
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../static/default-pic.png",
                  }
                }
                style={
                  Object {
                    "borderRadius": 28,
                    "height": 56,
                    "width": 56,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": null,
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 1,
                  "marginLeft": 15,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                James Cameroon
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                note={true}
                style={
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": null,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                Doing what you like will always keep you happy..
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "stretch",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flex": 0,
                  "paddingBottom": 10,
                  "paddingLeft": null,
                  "paddingRight": 15,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                note={true}
                style={
                  Object {
                    "alignSelf": null,
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "200",
                  }
                }
                uppercase={false}
              >
                2
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`main screen tests renders view correctly with defaults 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  viewIsInsideTabBar={false}
>
  <View>
    <View
      closeOnRowBeginSwipe={false}
      closeOnRowPress={true}
      closeOnScroll={true}
      directionalDistanceChangeThreshold={2}
      disableLeftSwipe={false}
      disableRightSwipe={false}
      leftOpenValue={0}
      previewFirstRow={false}
      recalculateHiddenLayout={false}
      rightOpenValue={0}
      style={Object {}}
      swipeToOpenPercent={50}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          avatar={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../static/default-pic.png",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": null,
                "alignSelf": null,
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 1,
                "marginLeft": 15,
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              uppercase={false}
            >
              James Bond
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              uppercase={false}
            >
              Doing what you like will always keep you happy..
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "stretch",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 0,
                "paddingBottom": 10,
                "paddingLeft": null,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "alignSelf": null,
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "200",
                }
              }
              uppercase={false}
            >
              0
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          avatar={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "center",
                "flex": 0,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../static/default-pic.png",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": null,
                "alignSelf": null,
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 1,
                "marginLeft": 15,
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              uppercase={false}
            >
              James Cameroon
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginLeft": null,
                  "marginRight": 10,
                }
              }
              uppercase={false}
            >
              Doing what you like will always keep you happy..
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "stretch",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flex": 0,
                "paddingBottom": 10,
                "paddingLeft": null,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              note={true}
              style={
                Object {
                  "alignSelf": null,
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "200",
                }
              }
              uppercase={false}
            >
              2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
